File: ONLINE_CODING_TEST/school.programmers/lvl_1/120808_분수의_덧셈.java
class Solution {
  // 두 수의 최대공약수(GCD) 구하기
  private int GCD(int a, int b) {
    return b == 0 ? a : GCD(b, a % b);
  }

  public int[] solution(int numer1, int denom1, int numer2, int denom2) {
    // 두 분수를 통분
    int numer = (numer1 * denom2) + (numer2 * denom1);
    int denom = denom1 * denom2;

    // 분자와 분모의 최대공약수 구하기
    int gcd = GCD(denom, numer);

    // 각각 최대공약수로 나눠 약분, 기약분수로 만들기
    numer = numer / gcd;
    denom = denom / gcd;

    return new int[] { numer, denom };
  }
}

File: ONLINE_CODING_TEST/school.programmers/lvl_1/120808_분수의_덧셈.js
// 두 수의 최대공약수 구하기
const GCD = (a, b) => (b === 0 ? a : GCD(b, a % b));

function solution(numer1, denom1, numer2, denom2) {
  // 두 분수를 통분
  let numer = numer1 * denom2 + numer2 * denom1;
  let denom = denom1 * denom2;

  // 분자와 분모의 최대공약수 구하기
  let gcd = GCD(numer, denom);

  // 각각 최대공약수로 나눠 약분, 기약분수로 만들기
  numer /= gcd;
  denom /= gcd;

  return [numer, denom];
}
